Automattic/wp-calypso
# Sites dashboard: styles fixes: header padding, footer removal unless there are children, search parent layout, and sites header font adjustments

Related to:
https://github.com/Automattic/automattic-for-agencies-dev/issues/64 (hiding bottom footer area)
https://github.com/Automattic/automattic-for-agencies-dev/issues/61 (search bar panel adjustments)
https://github.com/Automattic/automattic-for-agencies-dev/issues/60 (site list header padding)
https://github.com/Automattic/automattic-for-agencies-dev/issues/55 (sites table heading row)

## Proposed Changes

* This PR includes a few style changes:
  * The site footer top border is removed if there is nothing in the footer (in the site dashboard preview panel).
  * The site list header has padding adjustments to match design specs
  * The search bar panel now shows the search bar taking up most of the space, with the icons directly next to the bar.
  * The sites table header row has adjusted font size, font weight, color and line height, and the favorites icon has a modified vertical-align so it is more in line with the other headings.

See the Project Scope at pfunGA-w8-p2 for a link to the Figma i1 designs showing the above style changes.

## Testing Instructions


For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

**Footer**

* To replicate the issue, in trunk when visiting the sites dashboard, click on any site to open up the preview pane. You should see a very faint horizontal line further down the pane which represents the top of the footer.
* Apply this PR, and that line should no longer be visible (nor the footer code showing in the inspector).

**Site list header**

* This when is slightly harder to spot, but the padding around the Sites header and the search panel in particular is smaller to more closely match design specs (see link to Project Scope, which includes Figma link above)

**Search panel**

* In trunk, when opening up the preview pane the search bar itself is not filling up the empty space. There is a gap between the search field, then the filter icon, then another gap and the options icon.
* With this PR checked out, the search panel makes sure the search field fills up as much space as possible, with the other icons next to it.

**Sites table heading row**

* If you compare this row in trunk with what you see when the PR is checked out, the main thing you will notice is that the font sizes are smaller. There are several other CSS additions, to match what is specified in the issue linked above, plus a CSS change to make sure the favorites icon is more in line with the rest of the heading row items.

---

**Screenshots**

Search panel and sites header before (with preview pane open):
<img width="422" alt="Screenshot 2024-03-15 at 16 30 06" src="https://github.com/Automattic/wp-calypso/assets/16754605/ebf937e2-f2d3-4438-8ee1-ff823f0e8b1f">




Search panel and sites header after (with preview pane open):

<img width="413" alt="Screenshot 2024-03-15 at 16 28 51" src="https://github.com/Automattic/wp-calypso/assets/16754605/1df3aa1d-3e52-476d-b4c3-0c8c8ae91a77">

Sites table heading row before:

<img width="1476" alt="Screenshot 2024-03-15 at 17 04 54" src="https://github.com/Automattic/wp-calypso/assets/16754605/e7fe2ad6-0c09-4522-9163-7730c1e53010">


Sites table heading row after:


<img width="1536" alt="Screenshot 2024-03-15 at 18 08 55" src="https://github.com/Automattic/wp-calypso/assets/16754605/6f87eb1e-e1b5-40ab-8dae-0aba07f27e6a">

## Pre-merge Checklist
